### PR TITLE
Adjust some LMSFilePicker tests to await on components rendering when appropriate

### PR DESCRIPTION
**Pull request by acelaya**
This does not have user-facing or behavior changes, it just refactors a bit the `LMSFilePicker` tests so that they await on the "proper thing" (so to speak).

Some of those tests were "accidentally" passing just because there was an `await`, forcing the next tick, and consequently making some children components to be there.

However, as part of the changes in  #4853, this is not enough anymore.

Since the same pattern is present in a handful of tests, I have extracted the refactoring to a separated PR, to avoid polluting #4853, making it harder to review.

I have locally rebased that branch and it now works with these changes.

### CODE REVIEW

File: lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js [hunk 1]
```diff
@@ -184,16 +184,11 @@ describe('LMSFilePicker', () => {
 
     // Wait for the initial file fetch to fail.
     const wrapper = renderFilePicker();
-    try {
-      await fakeApiCall;
-    } catch {
-      /* unused */
-    }
     wrapper.update();
```
Comment (robertknight) [inline lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js:187]: This `wrapper.update()` should now be redundant, assuming `renderFilePicker` returns a "current" wrapper at the end, which is what I would expect as a reader.

File: lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js [hunk 2]
```diff
@@ -249,7 +238,6 @@ describe('LMSFilePicker', () => {
     // Make successful authorization attempt and wait for the auth window to close.
     fakeApiCall.reset();
     fakeApiCall.resolves([0]);
-    await fakeApiCall;
     wrapper.update();
```
Comment (robertknight) [inline lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js:241]: This `wrapper.update()` should also be redundant.

**Review by robertknight — APPROVED** [review]
Thanks for this! There are a couple of remaining `wrapper.update()` calls that I believe should be redundant. Otherwise looks good.

File: lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js (hunk 2 above)
Comment (acelaya) [reply lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js]: Fixed!

File: lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js (hunk 1 above)
Comment (acelaya) [reply lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js]: Fixed!

**Pull request merged**